### PR TITLE
Making some texts configurable

### PR DESCRIPTION
**Pull request by JOJ0**
Hi, thanks for your patience, I was deeply involved in another project the last few weeks, but finally here is the PR we talked about.

I made the texts "strings" and not objects (as you suggested including eg <Text> tags as well) for the simple reason that I couldn't get it to work right away. Let's call this a draft.

all the best
Jojo

### CODE REVIEW

File: lib/Root.js [hunk 1]
```diff
@@ -68,7 +68,10 @@ export default class Root extends React.Component {
 Root.defaultProps = {
   accessToken: null,
   filepath: null,
-  onError: () => false
+  onError: () => false,
+  loggedOutMsg: 'You can connect to Dropbox to backup/sync your transactions.',
+  loggedInMsg: 'You are connected to Dropbox, but you still need to select a file to use for sync.',
+  fileSeletctedMsg: 'You are connected to Dropbox and syncing to file:'
```
Comment (alexishevia) [inline lib/Root.js]: You have a typo here on the `fileSeletctedMsg` key. Should be `fileSelectedMsg`

File: lib/Root.js [hunk 2]
```diff
@@ -78,5 +81,8 @@ Root.propTypes = {
   onLogin: PropTypes.func.isRequired,
   onLogout: PropTypes.func.isRequired,
   onFilePick: PropTypes.func.isRequired,
-  onError: PropTypes.func
+  onError: PropTypes.func,
+  loggedOutMsg: PropTypes.string.isRequired,
+  loggedInMsg: PropTypes.string.isRequired,
+  fileSeletctedMsg: PropTypes.string.isRequired
```
Comment (alexishevia) [inline lib/Root.js]: Same typo here

**Review by alexishevia — CHANGES_REQUESTED** [review]
You're heading in the right direction, but there's stuff missing to make this work.

You have to pass along the messages from Root to the sub-components.

In these 3 places:
- FileSelected
   ```
        <FileSelected
          filepath={filepath}
          openFilePicker={openFilePicker}
          onLogout={onLogout}
          fileSelectedMsg={fileSelectedMsg} // <-- this is what you need to add
        />
   ```
   https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/Root.js#L54-L58
- LoggedIn
   ```
   <LoggedIn 
     openFilePicker={openFilePicker}
     onLogout={onLogout}
     loggedInMsg={loggedInMsg} // <-- this is what you need to add 
   />;
   ```
   https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/Root.js#L62
- LoggedOut
   ```
   <LoggedOut 
     appKey={appKey} 
     onLogin={onLogin} 
     onError={onError}
     loggedOutMsg={loggedOutMsg} // <-- this is what you need to add 
    />
   ```
   https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/Root.js#L64

**Comment by JOJ0** [comment]
Hi, sorry for forgetting this. I certainly had it when I coded/tested it in the project where I am actually using the FilePicker but forgot it when I put it in the freshly cloned FilePicker fork to send you the pull.

This happens if you don't test stuff..... :-/ I was to lazy to build a fresh rn project and put in the freshly adapted component. I am also not quite sure what would be the quickest workflow to create a dummy app, add the component (from local git) and code/test it. Started researching that right now.

One more thing: Will send you another commit soon. This also needs to be configurable to conclude this feature: https://github.com/alexishevia/rn-dropbox-filepicker/blob/master/lib/ChangeFileButton.js#L10

**Comment by JOJ0** [comment]
Well, I just learned that "yarn add" does understand any git repo urls and even local directories. That was too easy! Testing it right away issue solved...

File: README.md [hunk 3]
```diff
@@ -40,6 +40,9 @@ class YourComponent extends React.Component {
             onLogout={() => this.setState({ accessToken: null, filepath: null })}
             onFilePick={filepath => this.setState({ filepath })}
             onError={error => console.error(error.message)}
+            loggedInMsg={loggedInMsg}
+            loggedOutMsg={loggedOutMsg}
+            fileSelectedMsg={fileSelectedMsg}
```
Comment (alexishevia) [inline README.md:45]: Do you mind adding the `fileChangeWarningMsg` to the README?

Comment (alexishevia) [reply README.md:45]: Nevermind.. I'm creating a new PR to make it easier to contribute in the future. I'll add it there.

**Review by alexishevia — APPROVED** [review]
Thanks so much for this PR!

**Comment by JOJ0** [comment]
I have to thank you too! Learned a lot!

**Pull request merged**